.github/workflows/CI.yaml

```
@@ -70,10 +70,30 @@ jobs:
         pytest -v --cov=membrane_curvature --cov-report=xml --color=yes membrane_curvature/tests/
 
     - name: CodeCov
-      uses: codecov/codecov-action@v1
+      uses: codecov/codecov-action@v2
       with:
         file: ./coverage.xml
         flags: unittests
         name: codecov-${{ matrix.os }}-py${{ matrix.python-version }}
         fail_ci_if_error: True
 
+  latest_python:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        latest-python : ["3.11.0-rc.2"]
+    
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.latest-python }}
+
+    - name:  pip install membrane-curvature
+      run: |
+        python -m pip install . 
```

I always forget MDA is an upstream dependency. You're getting the failure because our pyproject.toml for 2.3.0 doesn't allow for a numpy variant that isn't in the 3.7-3.10 range. This does mean you probably won't be able to fully support py3.11 until MDA 2.4.0 gets released in a couple of month, but it would be a good idea for this job runner to essentially be the "cutting edge" job. i.e. installing the latest Python + the develop version of MDAnalysis. This would give you more certainty in the medium term longevity of the upcoming release (i.e. up until release the code worked with mda 2.4.0-dev0 and 3.11, so it would probably be fine when both get fully released).

To do that I would probably do:

```suggestion
        pip install git+https://github.com/MDAnalysis/mdanalysis.git@develop

        python -m pip install . 
```

---

Thanks, I was trying to find a version of MDA with Python 3.11 to install here. Just for future reference, where can I find the list of mda `X.Y.Z-dev` releases?

---

For PyPI maybe you mean: https://pypi.org/project/MDAnalysis/#files

I'm not up to date on if we do nightly/weekly wheel builds as well.

---

We don't unfortunately (yet? I'm still trying to think of ways not to annoy pypi / conda-forge with 40+ MB of tests every night/week).

---

For ~weekly, you can do something like NumPy/SciPy and use anaconda.org to host a pypi mirror like: https://anaconda.org/scipy-wheels-nightly/scipy

I think we just purge out the stale files once in a while, but apart from that you can use "normal" `pip` commands directed at the pypi mirror w/ `pip install -i <address of mirror>`. I know projects do that for cross-testing for breaks in dev.

It takes time to setup of course, but there's probably some stuff that can be copy-pasted from upstream for the uploads, apart from making our own MDA upload tokens for the mirror and storing them as secrets in the CI service or whatever.